src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,88 @@
+package duke;
+
+import duke.command.AliasCommand;
+import duke.command.Command;
+import duke.command.AddCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.ExitCommand;
+import duke.command.PlaceCommand;
+import duke.exception.DukeException;
+import duke.exception.EmptyFieldDukeException;
+import duke.exception.InvalidCommandDukeException;
+import duke.exception.PlaceParseDukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+public class Parser {
+
+    /**
+     * Looks at first token, determine & construct duke.command.Command sub-classes as per parsed.
+     *
+     * @param str Expects string to be parse as duke.command.Command.
+     * @return duke.command.Command object as per parsed.
+     * @throws DukeException if command can't be parsed.
+     */
+    public static Command parse(String str) throws DukeException {
+        assert str != null;
+
+        str = str.trim();
+        int lastIndex = str.indexOf(' ');
```

Comment:
This seems a little complicated. Can you achieve the same outcome by using .split?

Reply:
Definitely yes, I did a bit of hand-optimization here. As you can see, split is only good for one word. As you know split would process split at all white-spaces. But I just figured I could implement lazy look-ahead. Hence, great thought from you that I would improve. 👍